packages/repository/src/legacy-juggler-bridge.ts

```diff
@@ -136,6 +136,11 @@ export class DefaultCrudRepository<T extends Entity, ID>
     const model = await ensurePromise(
       this.modelClass.findById(id, filter, options),
     );
+    if (!model) {
+      return Promise.reject(
```

Comment:
FWIW, you can simply `throw new Error` since we are in an `async` function.

It would be good to include some machine-readable error code in the error reported by this function, so that the HTTP/REST layer can convert that code into HTTP status 404 Not Found.